sympy/printing/pretty/pretty.py

```diff
@@ -882,19 +882,18 @@ def _print_BlockMatrix(self, B):
         return self._print(B.blocks)
 
     def _print_MatAdd(self, expr):
-        s = None
-        for item in expr.args:
-            pform = self._print(item)
-            if s is None:
-                s = pform     # First element
-            else:
-                coeff = item.as_coeff_mmul()[0]
-                if S(coeff).could_extract_minus_sign():
-                    s = prettyForm(*stringPict.next(s, ' '))
-                    pform = self._print(item)
-                else:
-                    s = prettyForm(*stringPict.next(s, ' + '))
-                s = prettyForm(*stringPict.next(s, pform))
+        args = list(expr.args)
+        if len(args) == 0:
+            return None
+        s = self._print(args[0])
+        for item in args[1:]:
+            coeff = (
+                item.as_coeff_mmul()[0] if hasattr(item, 'as_coeff_mmul')
+                else item.as_coeff_mul()[0] if hasattr(item, 'as_coeff_mul')
+                else item)
+            joiner = ' ' if S(coeff).could_extract_minus_sign() else ' + '
```

Comment:
If `coeff` is not already sympified that is a bug elsewhere. So `S` is not (should not be) required. Probably better to not have it as it will hide bugs.

Reply:
Unfortunately that does not work because:

```python
[nav] In [10]: A = sympy.MatrixSymbol('A', 1, 1)

[ins] In [11]: B = sympy.MatrixSymbol('B', 1, 1)

[ins] In [12]: (A + B).args[1]
Out[12]: B

[ins] In [13]: (A + B).args[1].as_coeff_mmul()
Out[13]: (1, B)
```

Therefore:

```python
[ins] In [15]: coeff = (A + B).args[1].as_coeff_mmul()[0]

[ins] In [16]: coeff
Out[16]: 1

[ins] In [17]: coeff.could_extract_minus_sign()
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Input In [17], in <cell line: 1>()
----> 1 coeff.could_extract_minus_sign()

AttributeError: 'int' object has no attribute 'could_extract_minus_sign'
```

Reply:
Ahh, that is clearly a bug here:
https://github.com/sympy/sympy/blob/24afa468e69a9a50ed32c1d37f4a43528b99b575/sympy/matrices/expressions/matexpr.py#L400-L401
as it should not return 1, but S.One

I'll open a PR for that and changing check default to True.

We'll see what remains of the original issue then, but even though this may not be merged(?), it was clearly worthwhile to identify two issues in the code!